pkg/cmd/pulumi/main.go

```diff
@@ -45,9 +47,17 @@ func panicHandler() {
 
 func main() {
 	defer panicHandler()
+	initDebugGRPCTracing()
 	if err := NewPulumiCmd().Execute(); err != nil {
 		_, err = fmt.Fprintf(os.Stderr, "An error occurred: %v\n", err)
 		contract.IgnoreError(err)
 		os.Exit(1)
 	}
 }
+
+func initDebugGRPCTracing() {
+	if logFile := os.Getenv("PULUMI_DEBUG_GRPC"); logFile != "" {
+		err := interceptors.InitDebugInterceptors(logFile)
+		contract.AssertNoError(err)
+	}
+}
```

Comment:
If this were moved to rpcutil as a function like:

```go
func DebugInterceptors(debugMetadata) ([]grpc.UnaryClientInterceptor, []grpc.StreamClientInterceptor) {
  // if PULUMI_DEBUG_GRPC returns the interceptors
  // else return nil, nil
}
```

Could we change the usage to:

```go
  unaryDebugInterceptors, streamDebugInterceptors := rpcutil.DebugInterceptors(debugMetadata)
 
	conn, err := grpc.Dial(address, grpc.WithInsecure(),
		grpc.WithChainUnaryInterceptor(
			rpcutil.OpenTracingClientInterceptor(),
			unaryDebugInterceptors...
		),
		grpc.WithChainStreamInterceptor(
			rpcutil.OpenTracingStreamClientInterceptor(),
			streamDebugInterceptors...
		),
```

This would give us an extension point for future debugging modes & allows the logic to toggle this on or off to reside in one repository.

Reply:
That's nice, I can also give these tuples names (make them structs) so we're cleanly extensible on both sides.